src/HotChocolate/Core/src/Types/Types/OffsetPaging/ICollectionSlice.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+
+namespace HotChocolate.Types.OffsetPaging
+{
+    public interface ICollectionSlice
+    {
+        IReadOnlyCollection<object> Nodes { get; }
+
+        int TotalCount { get; }
```

Comment:
@ademchenko shall we move this into a separate interface `IPageInfo` like with the connection base paging?

Reply:
@michaelstaib, I didn't do that because that IPageInfo would contain just one property TotalCount instead of five in the case of cursor-based paging. Actually I don't think this is subject to change in the future so could we leave as is?

Reply:
The question is if we want to also introduce some indicator props that give us more info about the current segment like HasNextPage / HasPrevious page.

Anyhow, we will make it nullable since the `ICollectionSegmentResolver`can choose to not allow `totalCount` for performance reasons. The default will be to not have a total count. We also reworked the cursor pagination this way.